// @pactprotocol/mcp-client/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  PactApiClient: () => PactApiClient,
  PactClient: () => PactClient,
  createPactClient: () => createPactClient
});
module.exports = __toCommonJS(index_exports);

// src/client.ts
var import_client = require("@modelcontextprotocol/sdk/client/index.js");
var import_stdio = require("@modelcontextprotocol/sdk/client/stdio.js");
var PactClient = class {
  client;
  config;
  constructor(config) {
    this.config = {
      serverUrl: config.serverUrl || "https://mcp-server-production-f1fb.up.railway.app",
      rpcUrl: config.rpcUrl || "https://sepolia.base.org",
      ...config
    };
    this.client = new import_client.Client(
      {
        name: "pact-mcp-client",
        version: "0.1.0"
      },
      {
        capabilities: {}
      }
    );
  }
  /**
   * Connect to the Pact MCP server
   */
  async connect() {
    const transport = new import_stdio.StdioClientTransport({
      command: "node",
      args: ["--version"]
      // Placeholder
    });
    await this.client.connect(transport);
  }
  /**
   * Disconnect from the server
   */
  async disconnect() {
    await this.client.close();
  }
  /**
   * Call a tool on the MCP server
   */
  async callTool(name, args) {
    const result = await this.client.callTool({ name, arguments: args });
    return result;
  }
  /**
   * List available tools
   */
  async listTools() {
    return this.client.listTools();
  }
  /**
   * Get the underlying MCP client
   */
  getClient() {
    return this.client;
  }
};
function createPactClient() {
  const privateKey = process.env.PACT_WALLET_PRIVATE_KEY;
  if (!privateKey) {
    throw new Error("PACT_WALLET_PRIVATE_KEY environment variable is required");
  }
  return new PactClient({
    privateKey,
    serverUrl: process.env.PACT_MCP_SERVER_URL,
    rpcUrl: process.env.PACT_RPC_URL
  });
}

// src/api-client.ts
var PactApiClient = class {
  baseUrl;
  timeout;
  sessionId = null;
  constructor(options) {
    this.baseUrl = options.baseUrl.replace(/\/$/, "");
    this.timeout = options.timeout ?? 3e4;
  }
  /**
   * Set the session ID for authenticated requests
   */
  setSessionId(sessionId) {
    this.sessionId = sessionId;
  }
  /**
   * Get the current session ID
   */
  getSessionId() {
    return this.sessionId;
  }
  /**
   * Check if the server is healthy
   */
  async healthCheck() {
    try {
      const response = await this.fetch("/health", { method: "GET" });
      const data = await response.json();
      return data.status === "ok";
    } catch {
      return false;
    }
  }
  /**
   * List all available tools from the server
   */
  async listTools() {
    const response = await this.fetch("/tools", { method: "GET" });
    if (!response.ok) {
      throw new Error(`Failed to list tools: ${response.statusText}`);
    }
    const data = await response.json();
    return data.tools;
  }
  /**
   * Call a tool on the Pact MCP Server
   */
  async callTool(toolName, args = {}) {
    const response = await this.fetch(`/tools/${toolName}`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        ...this.sessionId && { "X-Session-Id": this.sessionId }
      },
      body: JSON.stringify(args)
    });
    const data = await response.json();
    if (!response.ok) {
      const error = data;
      const message = error.reason ? `${error.error}: ${error.reason}` : error.error;
      throw new Error(message);
    }
    return data;
  }
  /**
   * Internal fetch wrapper with timeout
   */
  async fetch(path, init) {
    const url = `${this.baseUrl}${path}`;
    const controller = new AbortController();
    const timeoutId = setTimeout(() => controller.abort(), this.timeout);
    try {
      const response = await fetch(url, {
        ...init,
        signal: controller.signal
      });
      return response;
    } catch (error) {
      if (error instanceof Error && error.name === "AbortError") {
        throw new Error(`Request to ${path} timed out after ${this.timeout}ms`);
      }
      throw error;
    } finally {
      clearTimeout(timeoutId);
    }
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PactApiClient,
  PactClient,
  createPactClient
});
